pkg/ottl/ottlfuncs/README.md

```diff
@@ -877,6 +878,26 @@ Examples:
 
 - `Int(Log(attributes["duration_ms"])`
 
+### MD5
+
+`MD5(value)`
+
+The `MD5` Converter converts the `value` to a md5 hash/digest.
+
+The returned type is string.
+
+`value` is either a path expression to a string telemetry field or a literal string. If `value` is another type an error is returned.
+
+If an error occurs during hashing it will be returned.
+
+Examples:
+
+- `MD5(attributes["device.name"])`
+
+- `MD5("name")`
+
+**Note:** According to the National Institute of Standards and Technology (NIST), MD5 is no longer a recommended hash function. It should be avoided except when required for compatibility. New uses should prefer FNV whenever possible.
```

Comment:
```suggestion
**Note:** According to the National Institute of Standards and Technology (NIST), MD5 is no longer a recommended hash function. It should be avoided except when required for compatibility. New uses should prefer a SHA-2 family function (e.g. SHA-256, SHA-512) whenever possible.
```
I'm not sure exactly what to recommend, but FNV is a non-cryptographic hash function, so we should recommend a different cryptographic hash function instead. OTTL at least has SHA-256, and there is an open PR for SHA-512.

Reply:
I see now that this was just copied from our existing hash functions, sorry for the trouble there. I've updated our other hash function notes as well: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34359